### PR TITLE
Jonmv/shut down transport

### PR DESCRIPTION
**Pull request by jonmv**
@havardpe please review

### CODE REVIEW

File: service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java [hunk 1]
```diff
@@ -76,6 +93,11 @@ public void applicationRemoved(ApplicationId id) {
     public void bootstrapComplete() {
     }
 
+    @Override
+    public void deconstruct() {
+        transport.sync().shutdown().join();
```
Comment (havardpe) [inline service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java:98]: extra sync up front should not be needed

**Review by havardpe — APPROVED** [review]

File: service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java [hunk 2]
```diff
@@ -30,14 +33,28 @@
...
 
     @Inject
     public SlobrokMonitorManagerImpl(DuperModelManager duperModel) {
-        this(SlobrokMonitor::new, duperModel);
+        this(new Transport("slobrok-monitor", getTransportThreadCount() / 4), duperModel);
     }
 
-    SlobrokMonitorManagerImpl(Supplier<SlobrokMonitor> slobrokMonitorFactory, DuperModelManager duperModel) {
+    private SlobrokMonitorManagerImpl(Transport transport, DuperModelManager duperModel) {
+        this(transport, new Supervisor(transport), duperModel);
```
Comment (havardpe) [inline service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java:48]: should the orb be tuned for small buffers?

File: service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java (hunk 1 above)
Comment (jonmv) [reply service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java:98]:
```suggestion
        transport.shutdown().join();
```

File: service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java (hunk 2 above)
Comment (jonmv) [reply service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java:48]: Perhaps. Ok.

File: service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java [hunk 3]
```diff
@@ -30,14 +33,28 @@
...
+    }
+
+    private SlobrokMonitorManagerImpl(Transport transport, Supervisor orb, DuperModelManager duperModel) {
+        this(() -> new SlobrokMonitor(orb), transport, duperModel);
```
Comment (jonmv) [inline service-monitor/src/main/java/com/yahoo/vespa/service/slobrok/SlobrokMonitorManagerImpl.java:52]:
```suggestion
        this(() -> new SlobrokMonitor(orb), transport, duperModel);
        orb.useSmallBuffers();
```

**Pull request merged**